logger_producer.go

```diff
@@ -0,0 +1,46 @@
+package timber
+
+import (
+	"context"
+
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/grip/send"
+	"github.com/pkg/errors"
+)
+
+// JasperLoggerOptions wraps LoggerOptions and implements a jasper
+// options.LoggerProducer for BuildloggerV3.
+type JasperLoggerOptions struct {
+	Ctx           context.Context `json:"-" bson:"-"`
```

Comment:
while I think there are situations where it's... possible to stash a context in a struct, it's generally a sign of code smell, and I think should be avoided in public interfaces. 

I think the options here are: 

- change the interface so that configure takes a context... this might generally be good, and is closer to what we mean. 
- change this to be a function which returns a context and a cancelFunc.